lib/grandpa/message_handler.go

```diff
@@ -562,6 +608,17 @@ func (s *Service) VerifyBlockJustification(hash common.Hash, justification []byt
 		return ErrMinVotesNotMet
 	}
 
+	if err := verifyBlockHashAgainstBlockNumber(s.blockState, fj.Commit.Hash, uint(fj.Commit.Number)); err != nil {
```

Comment:
split the function call from the error check

Reply:
why?

Reply:
The function takes a lot of arguments and it ends up being large. In other parts of the code, you use the separated call and check tho.

Reply:
done